exp/controllers/machinepool_controller_noderef.go

```diff
@@ -95,6 +97,12 @@ func (r *MachinePoolReconciler) reconcileNodeRefs(ctx context.Context, cluster *
 	mp.Status.UnavailableReplicas = mp.Status.Replicas - mp.Status.AvailableReplicas
 	mp.Status.NodeRefs = nodeRefsResult.references
 
+	if mp.Status.Replicas != mp.Status.ReadyReplicas {
+		conditions.MarkFalse(mp, expv1.ReplicasReadyCondition, expv1.WaitingForReplicasReadyReason, clusterv1.ConditionSeverityInfo, "")
+	} else {
+		conditions.MarkTrue(mp, expv1.ReplicasReadyCondition)
+	}
```

Comment:
Maybe I should avoid this one, since the pre-reqs to mark the machinepool ready is as described on line 56. Any thoughts?

Reply:
Seems like 56 covers the case where we have already reconciled these machines, but in case we need to do it again, we should definitely recalculate the final state